crates/rolldown/src/ast_scanner/impl_visit.rs

```diff
@@ -144,6 +144,7 @@ impl<'me, 'ast: 'me> Visit<'ast> for AstScanner<'me, 'ast> {
         self.result.self_referenced_class_decl_symbol_ids.insert(symbol_id);
       }
     }
+    self.check_import_assign(ident);
```

Comment:
Do you think what this PR does is similar to https://github.com/rolldown/rolldown/blob/bc16ada28c7122bf0246e8e3f46bb180179d6729/crates/rolldown/src/ast_scanner/impl_visit.rs#L183? Maybe we could handle them in one place?

Reply:
Maybe they are not the same thing

Reply:
https://hyrious.me/esbuild-repl/?version=0.23.0&b=e%00entry.js%00import+*+as+ns+from+%27c%27%0Ans.test+%3D+1%0A%0Aconst+a+%3D+1%3B%0Aa.test+%3D+100%3B&b=%00dep.js%00export+const+dep+%3D+1000%3B&o=%7B%0A++treeShaking%3A+true%2C%0A++external%3A+%5B%22c%22%2C+%22a%22%2C+%22b%22%5D%2C%0A%22bundle%22%3A+true%2C%0Aformat%3A+%22esm%22%2C%0Aplatform%3A+%22node%22%2C%0Aminify%3A+false%0A%7D

Reply:
here is an example, forbid_const_assign don't care object property assgin, but for a `ns` import it needs to be reported.

Reply:
Seems just a special pattern of forbidding assign.